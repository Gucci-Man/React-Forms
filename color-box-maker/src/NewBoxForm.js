import React, {useState} from 'react';

const NewBoxForm = ({ addBox }) => {
    const INITIAL_STATE = {
        color:  '',
        box_height: '',
        box_width: ''
    }
    const [formData, setFormData] = useState(INITIAL_STATE);
    const handleChange = (e) => {
        let { name, value } = e.target;
        // If value is a integer, convert to integer
        if(parseInt(value)) {
            value = parseInt(value);
        }
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        addBox({...formData});
        setFormData(INITIAL_STATE);
    }
    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="color">Background Color</label>
            <input 
                id="color"
                type="text"
                name="color"
                placeholder="Your background color"
                value={formData.color}
                onChange={handleChange}
            />
            <label htmlFor="box_height">Box Height</label>
            <input 
                id="box_height"
                type="text"
                name="box_height"
                placeholder="Your box height"
                value={formData.box_height}
                onChange={handleChange}
            />
            <label htmlFor="box_width">Box Width</label>
            <input 
                id="box_width"
                type="text"
                name="box_width"
                placeholder="Your box width"
                value={formData.box_width}
                onChange={handleChange}
            />
            <button>Add Box</button>
        </form>
    )
}

export default NewBoxForm;