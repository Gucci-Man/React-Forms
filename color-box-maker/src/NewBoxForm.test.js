import React from 'react';
import NewBoxForm from './NewBoxForm';
import { render } from "@testing-library/react";

// Smoke test
it("renders without crashing", function () {
  render(<NewBoxForm />);
});

// Snap test
it("matches snapshot", function () {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
  });