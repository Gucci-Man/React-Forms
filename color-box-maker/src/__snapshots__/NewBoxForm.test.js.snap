// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="color"
    >
      Background Color
    </label>
    <input
      id="color"
      name="color"
      placeholder="Your background color"
      type="text"
      value=""
    />
    <label
      for="box_height"
    >
      Box Height
    </label>
    <input
      id="box_height"
      name="box_height"
      placeholder="Your box height"
      type="text"
      value=""
    />
    <label
      for="box_width"
    >
      Box Width
    </label>
    <input
      id="box_width"
      name="box_width"
      placeholder="Your box width"
      type="text"
      value=""
    />
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
